### PR TITLE
Add `define_vector` function to `FusionDefinition` in Python API

### PR DESCRIPTION
**Pull request by kevinstephano**
Adding the `fd.define_vector()` function to the `FusionDefinition`. 

This includes the underlying `VectorRecord` but does not include the slightly different record for consuming `Scalar`s and producing a `Vector`.  This will occur in a future PR. 
1.  `fd.define_vector(size)` -- allows you to define an input vector which is really just a bunch of `Scalar`s.
2. `fd.define_vector([2, 2, 3])` -- allows you to define a constant `Vector` of integers.

I did not bump this `version` because this PR is not meant to be used, by itself, but does not hurt any other functionality.

### CODE REVIEW

**Comment by kevinstephano** [comment]
!build

**Comment by kevinstephano** [comment]
!build

File: csrc/serde/fusion_cache.fbs [hunk 1]
```diff
@@ -106,6 +106,8 @@ enum RecordType : int {
     TensorSizes,
     VarianceOp,
     VarianceMeanOp,
+    VectorInput,
```
Comment (rdspring1) [inline csrc/serde/fusion_cache.fbs:109]: What about naming the vector options `VectorDynamic` and `VectorStatic`?

**Review by rdspring1 — APPROVED** [review]
LGTM.

**Comment by kevinstephano** [comment]
!build

**Pull request merged**